hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenCancelAirdropHandler.java

```diff
@@ -126,15 +134,127 @@ public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckExcept
         }
     }
 
+    @SuppressWarnings("java:S3864")
     @Override
     public void handle(@NonNull final HandleContext context) throws HandleException {
-        var tokensConfig = context.configuration().getConfigData(TokensConfig.class);
+        requireNonNull(context);
+        final var txn = context.body();
+        final var op = txn.tokenCancelAirdropOrThrow();
+        configValidation(context.configuration(), op);
+
+        final var accountStore = context.storeFactory().writableStore(WritableAccountStore.class);
+        final var airdropStore = context.storeFactory().writableStore(WritableAirdropStore.class);
+
+        final var pendingAirdropIds = op.pendingAirdrops();
+        final var payer = context.payer();
+        final var payerAccount = getIfUsable(payer, accountStore, context.expiryValidator(), INVALID_ACCOUNT_ID);
+        validateTrue(payerAccount.hasHeadPendingAirdropId(), INVALID_TRANSACTION_BODY);
+
+        validatePendingAirdropIds(context, pendingAirdropIds, payer, accountStore, airdropStore);
+        deleteAccountAirdropsFromAccount(pendingAirdropIds, airdropStore, payerAccount, accountStore);
```

Comment:
⚠️ This changes the `payerAccount`, so in next step `updateAccountsNumberOfPendingAirdrops` we should fetch the payerAccount again, otherwise it will not have the previous changes. It is always safe to just pass in `payerId` and get from account store in each method instead of passing int `payerAccount` object.

Reply:
Thank you!!
I was stuck with a problem yesterday and this turns out to be the problem! Fixed 🔝